src/test/datascience/jupyter/kernels/installationPrompts.vscode.test.ts

```diff
@@ -246,20 +254,32 @@ suite('DataScience Install IPyKernel (slow) (install)', function () {
             'No errors in cell (third time)'
         );
     });
-    test('Ensure ipykernel install prompt is displayed every time you try to run a cell in an Interactive Window', async function () {
+    test('Ensure ipykernel install prompt is displayed every time you try to run a cell in an Interactive Window and you can switch kernels', async function () {
         if (IS_REMOTE_NATIVE_TEST) {
             return this.skip();
         }
+        // This is a complex test that tests the following workflow (which used to fail)
+        // Verify errors are displayed against the cells being executed.
+        // 1. Run cell against env that does not have ipykernel
+        // 2. Ensure we get a prompt
+        // 3. Dismiss the prompt (at this point we'd get yet another prompt).
+        // 4. Run another cell & we should get another prompt (this didn't work)
+        // 5. Run another cell & we should get another prompt (this didn't work)
+        // 6. Run another cell & select another kernel without ipykernel
+        // 7. We should get a prompt, however we used to get prompt for the previous env
```

Comment:
7/8. We should get two prompts: one for the original env, and one for the new one we switched to